### PR TITLE
Mac autobuild: Update to Xcode 12

### PR DESCRIPTION
**Pull request by hoffie**
Let the autobuild default to latest xcode (= 12.x) again, but still ensure that we limit SDK support to 10.15 due to lack of Qt-wise support for newer versions.

This is a different approach to the already merged #1655. This change has a (low) chance of fixing recent Mac issues on Big Sur 
(#1643, #1791). At least, it makes us use the latest officially recommended setup.

This only touches the regular Mac build, not the legacy build.

Support matrix:
https://doc.qt.io/qt-5/macos.html

Variables:
https://doc.qt.io/qt-5/qmake-variable-reference.html#qmake-mac-sdk

Explanations:
https://doc.qt.io/qt-5/macos-deployment.html

This PR is just a blind test if it works like that at all (cannot try locally). If it does, it would be interesting to have affected users test this version.

cc @softins

### CODE REVIEW

**Comment by hoffie** [comment]
Autobuild failed, `QMAKE_MAC_SDK=macosx10.15` doesn't seem to be respected while the other option (`QMAKE_MACOSX_DEPLOYMENT_TARGET=10.13`) is. Investigating.

**Comment by softins** [comment]
I think what you have done is not enough. Xcode 12 only includes the 11.1 SDK by default. See my test run from 25 days ago at https://github.com/softins/jamulus/runs/2522272216 where I had added a line `xcodebuild -sdk -version` to the prepare scripts to see which SDKs were present. There is only the latest one SDK for each platform.

You would need to determine what commands are needed to load an additional older SDK before you can select it for building with. I started to look at that, but then side-stepped it by switching to Xcode 11.

**Comment by hoffie** [comment]
> I think what you have done is not enough. Xcode 12 only includes the 11.1 SDK by default. See my test run from 25 days ago at https://github.com/softins/jamulus/runs/2522272216 where I had added a line `xcodebuild -sdk -version` to the prepare scripts to see which SDKs were present. There is only the latest one SDK for each platform.

Yep, thanks, I noticed that some minutes ago and tried Xcode 12 which still ships 10.15. Currently fails in python scripts, looking into this now.

> You would need to determine what commands are needed to load an additional older SDK before you can select it for building with. I started to look at that, but then side-stepped it by switching to Xcode 11.

Yeah, I think this was and is the simplest solution. It's just that I'm hoping that code generated by newer Xcode might magically make our leak bugs go away...

**Comment by softins** [comment]
If you are pinning the Xcode version to 12.0 (I think 12.1 would work too), in order to get the 10.15 SDK as default, I suspect you don't need the `EXTRA_QMAKE_ARGS` any more?

**Comment by softins** [comment]
https://en.wikipedia.org/wiki/Xcode#Xcode_11.x_-_12.x_(since_SwiftUI_framework)

**Comment by hoffie** [comment]
> If you are pinning the Xcode version to 12.0 (I think 12.1 would work too)

Ah, right. Trying with 12.1.1 now.

> in order to get the 10.15 SDK as default, I suspect you don't need the `EXTRA_QMAKE_ARGS` any more?

Hrm, yep, can probably drop it. Will leave it in now.

Do you know a way to use older SDKs with latest Xcode? All I've found was inofficial binary dumps, which I'd rather not use due to legal and security questions...?

**Comment by ann0see** [comment]
@emlynmac since you‘re a macOS specialist, could you have a look at this please?

**Comment by emlynmac** [comment]
Is there a reason to use 12.1.1 instead of the current 12.4 Xcode?

**Comment by ann0see** [comment]
I doubt there’s a reason. Probably 12.1.1 was the latest version as this PR was opened?

**Comment by emlynmac** [comment]
Maybe let's get as far up the chain as we can then?

**Comment by ann0see** [comment]
Yes, I think @dcorson-ticino-com worked on Qt 6 support, but I'm not sure how far he got.
So we should move this one up to the latest Xcode version.

File: .github/workflows/autobuild.yml [hunk 1]
```diff
@@ -143,13 +145,13 @@ jobs:
...
-      # For Qt5 on Mac, we need to ensure SDK 10.15 is used, and not SDK 11.x
-      # This is done by selecting Xcode 11.7 instead of the latest default of 12.x
+      # For Qt5 on Mac, we need to ensure SDK 10.15 is used.
+      # Xcode 12.0 is the last version which ships SDK 10.15, therefore pin it:
       - name:                       Select Xcode version for Mac
         if:                         ${{ matrix.config.target_os == 'macos' }}
         uses:                       maxim-lobanov/setup-xcode@v1
         with:
-          xcode-version: '11.7'
+          xcode-version: '12.1.1'
```
Comment (ann0see) [inline .github/workflows/autobuild.yml:154]:
```suggestion
          xcode-version: '12.4.0'
```

**Review by ann0see — CHANGES_REQUESTED** [review]

**Comment by ann0see** [comment]
Let's have a look at the CI

**Comment by dcorson-ticino-com** [comment]
> 
> 
> Yes, I think @dcorson-ticino-com worked on Qt 6 support, but I'm not sure how far he got.
> So we should move this one up to the latest Xcode version.

Yes, I have a Qt6 branch that compiles and works fine as far as I can see.
As I don't know what Xcode is I don't think I changed it.

**Comment by softins** [comment]
> Is there a reason to use 12.1.1 instead of the current 12.4 Xcode?

@emlynmac  Yes, see the comments further up (I've not been active over the weekend, so just picking up this conversation). Qt5 is only compatible with SDKs up to 10.15. If you use SDK 11, you get lots of warning messages from Qt about compatibility.

Xcode 12.1.1 is apparently the last one that comes with SDK 10.15, which is why we pinned that one. We should continue to do so, unless someone can come up with a way to use SDK 10.15 with later versions of Xcode.

**Comment by ann0see** [comment]
@dcorson-ticino-com could you please open a PR on your work? Probably all the macOS issues are somehow related and Qt6 might solve them.

@emlynmac @softins is it worth to move this forward for the next release or not?

**Comment by softins** [comment]
> @dcorson-ticino-com could you please open a PR on your work? Probably all the macOS issues are somehow related and Qt6 might solve them.

Certainly worth trying, but I'm not optimistic that Qt6 would be any better than Qt5 on Mac.

> @emlynmac @softins is it worth to move this forward for the next release or not?

Maybe for 3.9.0, but certainly not for 3.8.1.

**Comment by dcorson-ticino-com** [comment]
> @dcorson-ticino-com could you please open a PR on your work? Probably all the macOS issues are somehow related and Qt6 might solve them.

Give me a couple of days, I'm no longer sure in what kind of state it is in.

**Comment by ann0see** [comment]
Ok. De-tagged.

**Comment by ann0see** [comment]
> Give me a couple of days, I'm no longer sure in what kind of state it is in.

@emlynmac @dcorson-ticino-com @pljones not quite sure what to do with this. I'd rather open an issue to discuss the macOS problems...

**Comment by dcorson-ticino-com** [comment]
> not quite sure what to do with this. I'd rather open an issue to discuss the macOS problems...

Tony has gone forward with this, but having compilation problems on git.  I had it working OK under both QT5 and QT6 compiled locally.
As to the macOS problems I think it will be worth while to see how QT6 helps.

**Comment by ann0see** [comment]
Ok. If we do remove the license note, we could get it to work.

**Comment by ann0see** [comment]
https://github.com/ann0see/jamulus/runs/4685522957?check_suite_focus=true

**Comment by hoffie** [comment]
@softins @ann0see @emlynmac What's the current state of this PR?
I think the original proposal (update to 12.1.1) still makes sense.

The initial PR did this, but it has been modified to update to 12.4, which is sub-optimal, I think:

This official page confirms that 12.1.x is (and probably will stay?) the last version to support SDK 10.15, which we need for Qt5 compatibility: https://developer.apple.com/support/xcode/
This inofficial page lists 12.1.1 as the most-recent 12.1.x release: https://xcodereleases.com/

As all the newly introduced logic wrt `EXTRA_QMAKE_ARGS` as its no longer needed, I'm closing this PR and will open a new one for the 12.1.1 update.

**Pull request closed without merging**